lib/mayaHydra/hydraExtensions/sceneIndex/mayaHydraSceneIndex.cpp

```diff
@@ -387,18 +388,32 @@ MayaHydraSceneIndex::MayaHydraSceneIndex(
 }
 
 MayaHydraSceneIndex::~MayaHydraSceneIndex()
+{
+    //If you get a crash in a callback with a nullptr for _sceneIndex, 
+    // it may be due to the fact that the _sceneIndex pointer has been nulled as its ref count reached 0 but the destructor is still being called.
+    //You should call RemoveCallbacksAndDeleteAdapters(); before the destructor is called.
```

Comment:
This is a comment for potential future debugging. I am not sure if this is still valid since I made changes since then, but I would prefer to leave the comment to save time later if we get a crash in callbacks while this class has been deleted.

Reply:
I'm with @debloip-adsk on this, it seems that there is a lifescope management issue in there.

Reply:
Now logged as a separate future task, resolving this conversation.